apps/passport-client/src/service-worker.ts

```diff
@@ -10,36 +10,57 @@ const impermanentCache = [
   // "/",
   // "/favicon.ico",
   // "/index.html",
-  // "/global-pcdpass.css",
   // "/global-zupass.css",
-  // "/js/index.js",
+  // "/js/index.js"
 ];
 
-async function addResourcesToCache(resources: string[]): Promise<void> {
+const permanentCache = [
+  "/semaphore-artifacts/16.wasm",
+  "/semaphore-artifacts/16.zkey",
+  "/artifacts/zk-eddsa-event-ticket-pcd/circuit.wasm",
+  "/artifacts/zk-eddsa-event-ticket-pcd/circuit.zkey",
+  "/fonts/IBMPlexSans-Regular.ttf",
+  "/fonts/IBMPlexSans-Medium.ttf",
+  "/fonts/IBMPlexSans-Light.ttf",
+  "/fonts/IBMPlexSans-ExtraLight.ttf",
+  "/fonts/IBMPlexSans-Regular.woff",
+  "/fonts/IBMPlexSans-Medium.woff",
+  "/fonts/IBMPlexSans-Light.woff",
+  "/fonts/IBMPlexSans-ExtraLight.woff"
+];
+
+async function addResourcesToCache(): Promise<void> {
   const cache = await caches.open("v1");
 
+  const keys = await cache.keys();
+
+  // Delete all existing cache entries not in our "permanent" list.
+  // This eliminates any stale entries from old service workers which
+  // used a different list.
   await Promise.all(
-    ["/", "/index.html", "/js/index.js"].map((item) => cache.delete(item))
+    keys.map((request: Request) => {
+      if (!permanentCache.includes(new URL(request.url).pathname)) {
```

Comment:
the conversion `new URL(request.url).pathname` feels like it could be extracted to a function, perhaps one named `requestToItemCacheKey` or something similar. another thing is that it seems a little brittle - eg. what happens when we add a cached file with a particular query string? tl;dr the `.pathname` thing is not an isomorphism which scares me a bit

Reply:
All this URL handling is black magic to me anyway, but at least we know what the entries in our list are, and they're simple.  I'll make the helper function, and we can worry about making it handle URL parameters if we ever need that.